lib/rucio/web/ui/common/utils.py

```diff
@@ -20,6 +20,51 @@
 from rucio.common.config import config_get
 from rucio.db.sqla.constants import AccountType
 
+try:
+    AUTH_TYPE = config_get('webui', 'auth_type')
+    if AUTH_TYPE == 'saml':
+        from onelogin.saml2.auth import OneLogin_Saml2_Auth
+except:
+    AUTH_TYPE = 'x509'
+
+
+def prepare_webpy_request(request, data):
+    """
+    Prepare a webpy request for SAML
+
+    :param request: webpy request object
+    :param data: GET or POST data
+    """
+    return {
+        'https': 'on' if request['wsgi.url_scheme'] == 'https' else 'off',
+        'http_host': request['HTTP_HOST'],
+        'server_port': request['SERVER_PORT'],
+        'script_name': request['SCRIPT_NAME'],
+        'get_data': data,
+        # Uncomment if using ADFS as IdP
+        # 'lowercase_urlencoding': True,
+        'post_data': data
+    }
+
+
+def set_cookies(token, cookie_accounts, attribs, ui_account=None):
+    # if there was no valid session token write the new token to a cookie.
+    if token:
+        setcookie('x-rucio-auth-token', value=token, path='/')
```

Comment:
cookies are case-sensitive, and for consistency we should keep them in line with the headers